### PR TITLE
Fix schema content deprications

### PR DESCRIPTION
**Pull request by YvarRavy**
described in: #5692

i've basically moved the logic from Content class to the ContentHandler. also i've changed a few variables from private to public so the ContentHandler can call them. i hope this is okay. this seemed to be the quickest way to fix the deprecation warning.

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling and processing logic for schema content rendering.
	- Improved visibility of certain properties and methods for greater accessibility.

- **Bug Fixes**
	- Removed deprecated method for populating parser output, streamlining content handling.

- **Tests**
	- Removed outdated test methods related to the deprecated `fillParserOutput` functionality.

### CODE REVIEW

**Comment by YvarRavy** [comment]
i forgot to change the tests, my bad. one problem i can move the tests from schemaContent to schemaContentHandler but since the fillParserOuput is protected in that class i can't call it. i could use the fillParserOutputInternal but i don't know if that is smart

**Comment by gesinn-it-gea** [comment]
Regarding testing etc. I think rewriting the logic to use setter/getter would be a better approach

**Comment by YvarRavy** [comment]
> Regarding testing etc. I think rewriting the logic to use setter/getter would be a better approach

Yeah i was planning todo that after seeing the response from the bot. But that doesn't fix the problem regarding the test case or do i not fully understand what you said?

**Comment by gesinn-it-gea** [comment]
... just wanted to give my 5 cents to why using getters / setters is better in the context of testing:

Using setters and getters instead of public variables provides encapsulation, allowing you to add validation, logging, or debugging checks directly within these methods. This control ensures that data changes adhere to specific rules, which is particularly useful in testing scenarios. With public variables, you lose this ability, risking inconsistent or invalid states that can lead to hard-to-diagnose bugs. Setters/getters also make future modifications easier (e.g., adding validation) without changing the class interface, keeping tests stable.

This should to my opinion be considered when code and tests are changed.

**Comment by JeroenDeDauw** [comment]
Agreed. Seems like you are only reading the property, so a public getter is better than making the property public, since a getter does not allow for mutation.

**Comment by YvarRavy** [comment]
> Agreed. Seems like you are only reading the property, so a public getter is better than making the property public, since a getter does not allow for mutation.

do you have an idea on what todo with the unitTests? as said before i can move them but the fillParserOutput is protected inside the ContentHandler

**Comment by YvarRavy** [comment]
@JeroenDeDauw i'm not sure what todo with the unit tests, since the fillParserOutput is protected. can i just remove the unit tets?

File: src/MediaWiki/Content/SchemaContent.php [hunk 1]
```diff
@@ -268,6 +268,24 @@ public function initServices() {
 		}
 	}
 
+	/**
+	 * Gets the content formatter.
+	 *
+	 * @return SchemaContentFormatter|null The content formatter instance or null if not set.
+	 */
+	public function getContentFormatter() {
```
Comment (JeroenDeDauw) [inline src/MediaWiki/Content/SchemaContent.php:276]: These comments do not add much value. You could just do 

```php
public function getContentFormatter(): SchemaContentFormatter|null {
```

File: tests/phpunit/MediaWiki/Content/SchemaContentTest.php [hunk 2]
```diff
@@ -245,109 +245,4 @@ public function testPreSaveTransform() {
...
-
-		$instance = new SchemaContent(
-			json_encode( [ 'Foo' => 42 ] )
-		);
-
-		$generateHtml = true;
-		$revId = 42;
-
-		$instance->fillParserOutput( $title, $revId, $parserOptions, $generateHtml, $parserOutput );
-	}
```
Comment (JeroenDeDauw) [inline tests/phpunit/MediaWiki/Content/SchemaContentTest.php:351]: Throwing out these poorly designed and brittle tests is reasonable enough. At the same time it's a pity to not have any good tests for the modified code.

**Pull request merged**